cmd/thanos/query.go

```diff
@@ -240,7 +247,11 @@ func runQuery(
 			func() (specs []query.StoreSpec) {
 				// Add DNS resolved addresses.
 				for _, addr := range dnsProvider.Addresses() {
-					specs = append(specs, query.NewGRPCStoreSpec(addr, false))
+					if dnsServerName {
+						specs = append(specs, query.NewGRPCStoreSpecServerName(addr, false, dnsProvider.ServerName(addr)))
```

Comment:
```suggestion
						specs = append(specs, query.NewGRPCStoreSpecServerName(addr, false, serverName)
```

can we just use serverName provided from `--grpc-client-server-name`  instead of relying on dns?

Reply:
Currently, the code as-is will use a provided server-name.  This works when using a single store from the querier.  But adding multiple store options to collect data from many stores will not be able to use individual server names as the code currently stands.

The change intended here is to allow for multiple stores to each use different server names.